plugins/SitesManager/SiteContentDetection/VueJs.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Plugins\SitesManager\SiteContentDetection;
+
+use Piwik\Container\StaticContainer;
+use Piwik\Piwik;
+use Piwik\SiteContentDetector;
+use Piwik\Url;
+use Piwik\View;
+
+class VueJs extends SiteContentDetectionAbstract
+{
+    public static function getName(): string
+    {
+        return 'Vue.js';
+    }
+
+    public static function getContentType(): string
+    {
+        return self::TYPE_JS_FRAMEWORK;
+    }
+
+    public static function getInstructionUrl(): ?string
+    {
+        return 'https://matomo.org/faq/new-to-piwik/how-do-i-install-the-matomo-tracking-code-on-websites-that-use-vue-js/';
+    }
+
+    public static function getPriority(): int
+    {
+        return 50;
+    }
+
+    public function detectByContent(?string $data = null, ?array $headers = null): bool
+    {
+        return preg_match("/vue\.\w.+.js|vue\-\w.+.js/i", $data) === 1;
+    }
+
+    public function shouldHighlightTabIfShown(): bool
+    {
+        return true;
+    }
+
+    public function renderInstructionsTab(SiteContentDetector $detector = null): string
+    {
+        $view     = new View("@SitesManager/_vueTabInstructions");
+        $view->sendHeadersWhenRendering = false;
+        $view->SiteWithoutDataVueFollowStepNote2Key = StaticContainer::get('SitesManager.SiteWithoutDataVueFollowStepNote2');
+        $view->vue3Code = $this->getVueInitializeCode(3);
+        $view->vue2Code = $this->getVueInitializeCode(2);
+        return $view->render();
+    }
+
+    public function renderOthersInstruction(SiteContentDetector $detector = null): string
+    {
+        if ($detector->wasDetected(self::class)) {
+            return ''; // don't show on others page if tab is being displayed
+        }
```

Comment:
This is repeating quite a few times, should the logic be in the Others template rather than in each of the detectors?

Reply:
Then it wouldn't be possible to show it when it was detected, which would be incorrect.
With the new layout that will be removed anyway I guess. As Vue and other stuff will be displayed always on the front page, there is no benefit in displaying it on the others page anymore.